src/content/blog/astro-250.mdx

```diff
@@ -0,0 +1,167 @@
+
+```js
+import { defineConfig } from "astro/config"
+import nodejs from "@astrojs/node"
+
+export default defineConfig({
+	output: "hybrid",
+	adapter: nodejs(),
+	experimental: {
+		hybridOutput: true,
+	},
+})
+```
+
+Once you've done that your static site can not _opt out_ of prerendering by setting the `prerender` export of a page or endpoint to `false`:
+
+**src/pages/contact.astro**
+
+```astro
+---
+export const prerender = false
+
+if (Astro.request.method === "POST") {
+	// handle form submission
+}
+---
+
+<form method="POST">
+	<input type="text" name="name" />
+	<input type="email" name="email" />
+	<button type="submit">Submit</button>
+</form>
+```
+
+## Custom client directives (experimental)
+
+Custom client directives allow integration authors to define new `client:` directives to gain greater control over when components load.
+
+Integrations can add these through the `astro:config:setup` hook's `addClientDirective()` API. To enable this API, the user needs to enable `customClientDirectives` in the `experimental` section of their config.
+
+**astro.config.mjs**
+
+```js
+import { defineConfig } from "astro/config"
+import onClickDirective from "astro-click-directive"
+
+export default defineConfig({
+	integrations: [onClickDirective()],
+	experimental: {
+		customClientDirectives: true,
+	},
+})
+```
+
+**astro-click-directive**
+
+```js
+export default function onClickDirective() {
+	return {
+		hooks: {
+			"astro:config:setup": ({ addClientDirective }) => {
+				addClientDirective({
+					name: "click",
+					entrypoint: "astro-click-directive/click.js",
+				})
+			},
+		},
+	}
+}
+```
+
+Now you can use `client:click` on any of your framework components with full type support.
+
+```astro
+<Counter client:click />
+```
+
+See the [LINK TO DOCS](#) to learn how to build these types of integrations.
+
+## HTML minification
+
+You can now opt-in to minification of the HTML produced by your Astro components.
+
+Using the `compressHTML` option Astro will remove all whitespace.
+
+We wanted to enable minification without sacrificing performance. For server-rendered apps, doing minification on each render could be costly. With `compressHTML` your components are compressed by the Astro compiler, only once and during the build.
+
+You can enable this in your config:
+
+```js
+import { defineConfig } from "astro/config"
+export default defineConfig({
+	compressHTML: true,
+})
+```
+
+Compression occurs both in development mode and in the final build.
+
+This option _will not_ compress HTML produced by framework components. If you'd like that functionality it can easily be achieved by writing a [middleware](https://docs.astro.build/en/guides/middleware/) to compress HTML responses.
+
+## Parallelized rendering
+
+When rendering components Astro will now render them in parallel, allowing components that do their own data loading to not be blocked by other components higher in the tree that are also loading data. A simple example:
```

Comment:
```suggestion
Astro will now render components in parallel so that data-loading components higher in the tree will no longer block other components that are also loading data, as in the example below:
```